### PR TITLE
refactor(test): reduce execKarma to a reasonable size

### PR DESCRIPTION
**Pull request by devoto13**
Finally!

Introduced extra step "I wait until server output contains" instead of hard-coded timeout. It will regularly evaluate a condition until it fulfills. This approach should potentially perform a little bit better than the previous solution.

### CODE REVIEW

File: test/e2e/step_definitions/utils.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+const sleep = promisify(setTimeout)
+
+module.exports.waitForCondition = async (evaluateCondition, timeout = 1000) => {
+  let remainingTime = timeout
+  while (!evaluateCondition()) {
+    if (remainingTime > 0) {
+      await sleep(50)
+      remainingTime -= 50
+    } else {
+      throw new Error('Condition not fulfilled until timeout')
```
Comment (johnjbarton) [inline test/e2e/step_definitions/utils.js:12]: I think this would be clearer as 'Timed out and condition not fulfilled'

Comment (devoto13) [reply test/e2e/step_definitions/utils.js:12]: I don't quite agree. The main point is that condition was not fulfilled, not that it timed out.

Comment (johnjbarton) [reply test/e2e/step_definitions/utils.js:12]: The `until` seems odd to me.  Maybe:
`Condition not fulfillled, waited ${timeout}ms`

Comment (devoto13) [reply test/e2e/step_definitions/utils.js]: Updated

**Pull request merged**